### PR TITLE
Enable ALL THE LINTS

### PR DESCRIPTION
**Pull request by Hixie**
Well, all the easy ones, anyway.

For some reason `// ignore:` isn't working for me so I've disabled
lints that need that. Also disabled those that require a ton of work
(which I'm doing, but not in this PR, to keep it reviewable).

This adds:
- avoid_init_to_null
- library_names
- package_api_docs
- package_names
- package_prefixed_library_names
- prefer_is_not_empty
- sort_constructors_first
- sort_unnamed_constructors_first
- unnecessary_getters_setters

### CODE REVIEW

**Comment by Hixie** [comment]
cc @abarth for review

fyi @chinmay for cassowary and newton
fyi @floitschG for scheduler
fyi @yjbanov for driver
fyi @vlidholt for markdown and sprites
fyi @devoncarew for tools

**Comment by Hixie** [comment]
oh and fyi @pq for all the lints!

**Comment by Hixie** [comment]
For the record, the lints I haven't yet enabled are:

- always_specify_types, which I'm working on doing on a separate branch and have already done about half of
- annotate_overrides, which I'll do next (this is just a mechanical change)
- avoid_as, which is blocked on https://github.com/dart-lang/linter/issues/195 making it through our system, but for which I've already done the relevant code changes in this PR
- avoid_return_types_on_setters, which is a mechanical change I'll do soon
- constant_identifier_names, which is blocked on getting `// ignore:` working
- implementation_imports, which is also blocked on `// ignore:` but for which I've already done the work in this PR
- one_member_abstracts, which is blocked on `// ignore:` but which is otherwise handled in this PR
- public_member_api_docs, because we haven't yet done ALL THE DOCS
- type_annotate_public_apis, which I believe is a subset of always_specify_types above

File: packages/flutter/lib/src/gestures/arena.dart [hunk 1]
```diff
@@ -100,7 +100,7 @@ class GestureArena {
       return;  // This arena is being held for a long-lived member
     }
     _arenas.remove(arenaKey);
-    if (!state.members.isEmpty) {
+    if (state.members.isNotEmpty) {
```
Comment (abarth) [inline packages/flutter/lib/src/gestures/arena.dart]: Oh, nice.

**Comment by abarth** [comment]
LGTM

File: packages/flutter/lib/src/gestures/arena.dart (hunk 1 above)
Comment (pq) [inline packages/flutter/lib/src/gestures/arena.dart]: Cheers to @hterkelsen for suggesting this check! :+1:

**Comment by pq** [comment]
Needless to say, :metal:

**Pull request merged**